rfcs/0020-security-on-call.md

```diff
@@ -0,0 +1,165 @@
+---
+feature: security-on-call
+start-date: 2017-10-30
+author: Graham Christensen
+co-authors: Franz Pletz
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+Organize and distribute the handling of public vulnerability
+disclosures, through the use of a community-based team and rotating
+"point" or "on-call" assignments.
+
+# Motivation
+[motivation]: #motivation
+
+## 1. Security Posture, Prompt Patching, and Documented Process
+
+Our process for security issues is currently fairly well executed, but
+porly defined. How to participate and how to do the job is very
+nebulous and boils down to:
+
+1. Everybody pays attention
+2. Everybody patches
+
+This system means there is no division of labor, and no documented way
+to ensure patches are patched the same way, every time.
+
+## 2. Community and Commitment Sensitivity
+
+As a community, we have a few dedicated contributors patching
+the vast majority of security issues. This has worked well, and has
+even ensured fairly good coverage for some time now.
+
+Security patching work is an easy way to burn contributors out. The
+perpetual feed of new issues to fix is exhausting. This is concerning,
+because most of the patches are being applieed by highly skilled,
+involved, and "core" members of the NixOS community. It would be a
+shame to lose them.
+
+At the same time, most security patches are easy to apply:
+
+ - Most announcements to security email lists include easy-to-apply
+   patches.
+ - Well established, well funded distributions have full time
+   employees focusing on security matters. These other distributions
+   regularly publish their minimal security patches on their bug
+   trackers or in their source trees.
+ - Many security issues can be easily fixed by minor package bumps.
+
+Some of the more tricky parts are determining how to backport the
+patch, and finding build capacity for testing large rebuild changes.
+Both of these questions can easily be answered by asking more
+experienced community members for help.
+
+# Detailed design
+[design]: #detailed-design
+
+I propose we create an on-call rotation for publicly disclosed
+security issues. Each member of the team will be on call for 24 hours,
+and expected to handle every new issue which is created within their
+on-call period.
+
+The patching team will not handle issues under embargo.
+
+## Patching Team
```

Comment:
I would not refer to that group of people as "patching team". I'd rather call them "incident managers" or something like that. The way I see it, the primary role of those volunteers is *not* to fix the actual security issue. Their role is to ensure that:

- no security issue is overlooked,
- everything is recorded properly in some bug tracking system, so that others can check the current state of affairs there,
- the process if fixing the issue is delegated to the package's maintainer
- maintainers are notified (and reminded) of these issues.

While I think it's fine for members of that on-call team to fix security issues themselves, I believe that the *documentation* part of the job is equally or even more important.